### PR TITLE
cnp-1131-acr-tasks

### PR DESCRIPTION
**Pull request by harmrai**
### JIRA link
https://tools.hmcts.net/jira/browse/CNP-1131

### Change description
This PR leverages on layers re-use to reduce the the docker image build time in CI. However, this project using a specific branch of the Jenkins pipeline, **it will have an effect only if you use the latest version of the Jenkins library**.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,24 +1,17 @@
...
-
-COPY package.json yarn.lock ./
-RUN curl -o- -L https://yarnpkg.com/install.sh | bash -s \
-    && export PATH="$HOME/.yarn/bin:$HOME/.config/yarn/global/node_modules/.bin:$PATH" \
+# ---- Base Image ----
+FROM hmcts.azurecr.io/hmcts/base/node/stretch-slim-lts-8 as base
+RUN export PATH="$HOME/.yarn/bin:$HOME/.config/yarn/global/node_modules/.bin:$PATH" \
     && yarn install --production \
     && yarn cache clean
+COPY package.json yarn.lock ./
```
Comment (dwaynebailey) [inline Dockerfile]: Not sure I understand this bit.  Surely we need the package.json and yarn.lock before we do the install?  And we've lost the yarn upgrade.

File: Dockerfile [hunk 2]
```diff
@@ -1,24 +1,17 @@
...
+COPY . .
+RUN yarn install
 
-COPY app.js server.js ./
-COPY app ./app
-COPY config ./config
-
-ENV PORT 3460
-
-HEALTHCHECK --interval=10s \
```
Comment (dwaynebailey) [inline Dockerfile]: Is the `HEALTHCHECK` part of the base image?

Comment (slyg) [reply Dockerfile]: No, however the AKS config should come with its own healthcheck (liveness probes) so this instruction might be redoundant.

Afaik we do not intend adding any healthcheck at the container level.

File: Dockerfile (hunk 1 above)
Comment (slyg) [reply Dockerfile]: I agree, just a line swap should fix it

**Pull request merged**